examples/scala/src/main/scala/StandaloneExample.scala

```diff
@@ -0,0 +1,125 @@
+     */
+    private def execAppLogic(): Future[Unit] = {
+        def loop(): Future[Unit] = {
+            createClient().flatMap(client => appLogic(client).andThen {
+                case _ => Try(client.close()) match {
+                    case Failure(e) =>
+                        Logger.log(
+                            Logger.Level.WARN,
+                            "glide",
+                            s"Encountered an error while closing the client: ${e.getCause}"
+                        )
+                    case _ => ()
+                }
+            }).recoverWith {
+                case e: CancellationException =>
+                    Logger.log(Logger.Level.ERROR, "glide", s"Request cancelled: ${e.getMessage}")
+                    Future.failed(e)
+                case e: Exception => e.getCause match {
+                    case e: ClosingException if e.getMessage.contains("NOAUTH") =>
+                        // If the error message contains "NOAUTH", raise the exception
+                        // because it indicates a critical authentication issue.
+                        Logger.log(Logger.Level.ERROR, "glide", s"Authentication error encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case e: ClosingException =>
+                        Logger.log(Logger.Level.WARN, "glide", s"Client has closed and needs to be re-created: ${e.getMessage}")
+                        loop()
+                    case e: TimeoutException =>
+                        // A request timed out. You may choose to retry the execution based on your application's logic
+                        Logger.log(Logger.Level.ERROR, "glide", s"Timeout encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case e: ConnectionException =>
+                        // The client wasn't able to reestablish the connection within the given retries
+                        Logger.log(Logger.Level.ERROR, "glide", s"Connection error encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case _ =>
+                        Logger.log(Logger.Level.ERROR, "glide", s"Execution error encountered: ${e.getCause}")
+                        Future.failed(e)
+                }
+            }
+        }
+
+        loop()
+    }
+
+    /**
+     * The entry point of the standalone example. This method sets up the logger configuration
+     * and executes the main application logic.
+     *
+     * @param args Command-line arguments passed to the application.
+     */
+    def main(args: Array[String]): Unit = {
+        // In this example, we will utilize the client's logger for all log messages
+        Logger.setLoggerConfig(Logger.Level.INFO)
+        // Optional - set the logger to write to a file
+        // Logger.setLoggerConfig(Logger.Level.INFO, file)
+
+        // Await is used only for this example. Not recommended for use in production environments.
```

Comment:
So why do we use it?

Reply:
I've updated the timeout to not be infinity so it can be used in a production environment.